python/drned_xmnr/op/parse_log_errors.py

```diff
@@ -127,59 +112,73 @@ def add_new_problem(problems: List[ProblemData], problem: ProblemData) -> None:
         problems.append(problem)
 
 
-# logger: List[str],
-def gather_problems(file: TextIO) -> List[ProblemData]:
-    """ Read through the input file and collect data for all the idetified
-        problems found inside. """
-    problems: List[ProblemData] = []
-
-    pending = PendingData()
-
-    for line_num, line in enumerate(file):
-        test_case = parse_test_case(line)
-        # new test-case opening
-        if test_case is not None:
-            if pending.problem is not None:
-                add_new_problem(problems, pending.problem)
-                pending.problem = None
-            pending.test_case = test_case
-            # logger.append("---- new test-case: line %d: %s" % (line_num + 1, test_case))
-            continue
-
-        # potential start of new problem
-        if pending.matcher is None:
-            pending.matcher = _get_matcher(line)
+class ProblemsParser:
+    def __init__(self, list_node) -> None:
+        self.matchers = []
+        for item in list_node:
+            match_start = item.match
+            match_stop = None if item.terminator is None else item.terminator
+            max_lines = 0 if item.max_lines is None else item.max_lines
+            self.matchers.append(ProblemMatcher(match_start=match_start, match_stop=match_stop, max_lines=max_lines))
+
+    def _get_matcher(self, line: str) -> Optional[ProblemMatcher]:
+        for m in self.matchers:
+            if m.match_start in line:
+                return m
+        return None
+
+    def gather_problems(self, file: TextIO) -> List[ProblemData]:
+        """ Read through the input file and collect data for all the idetified
+            problems found inside. """
+        problems: List[ProblemData] = []
+
+        pending = PendingData()
+
+        for line_num, line in enumerate(file):
+            test_case = parse_test_case(line)
+            # new test-case opening
+            if test_case is not None:
+                if pending.problem is not None:
+                    add_new_problem(problems, pending.problem)
+                    pending.problem = None
+                pending.test_case = test_case
+                # logger.append("---- new test-case: line %d: %s" % (line_num + 1, test_case))
```

Comment:
All these `logger.append` lines should be removed.

Reply:
noted & removed in pending PR commit...